Dockerfile

```diff
@@ -1,10 +1,34 @@
-# Goreleaser builds a docker image according to this Dockerfile by copying the sysl binary 
-# to a golang:1.13-alpine image and setting up the entrypoint.
+# Builds a docker image according to this Dockerfile by building the sysl binary
+# on Go 1.13 (by default) and then copying it
+# to a debian:buster-slim image and setting up the entrypoint.
 #
 # The produced image is published to https://hub.docker.com/r/anzbank/sysl
+#
+# To use this Docker file run the following command
+#    docker image build --build-arg sysl_tag=v0.7.1  -t my-sysl:0.7.1 .
+
+ARG go_ver=1.13
+ARG alpine_ver=3.11
+
+FROM golang:${go_ver}-alpine${alpine_ver} as builder
+
+ARG sysl_tag
+
+RUN test -n "${sysl_tag}" || ( echo Plese set sysl_tag build-arg && exit 1 )
 
-FROM golang:1.13-alpine
+RUN apk --no-cache add git
 
-COPY sysl /
+WORKDIR /sysl
+
+RUN git clone --branch ${sysl_tag} --single-branch --depth 1 https://github.com/anz-bank/sysl.git . && \
+    go install ./cmd/sysl
+
+FROM alpine:${alpine_ver} as runner
```

Comment:
"as runner" is unnecessary here. please remove.

As mentioned in one of my comments, `sysl` uses the `go` command for modules, so the image needs to be a `golang` one.

Using 3.11 as `${alpine_ver}` on the final image means that this is not a reproducible build - the 3.11 tag moves as additional 3.11 point releases are made. I prefer to use fixed tags for final images so that the image can be reproducibly built from the same commit.

Reply:
I use "as runner" to give the reader extra hints if ever extra layers are added. It seems like good hygiene and does no harm

Changed to golang

I prefer my images to pick up fixes so I don't usually provide the patch version. i.e. if Go get some fixes I want them the next time I build.

However I have changed to specify the full version as requested

Reply:
Yeah, I see both sides of this, but there are certain environments where reproducible builds are a requirement. We should consider using tools like renovate (https://github.com/renovatebot/renovate) to automatically bump the version so we get the best of both worlds.